pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -678,6 +678,11 @@ func (ctrl *Controller) syncImageConfig(key string) error {
 		}
 	}
 
+	// append the internalRegistryHostname to allowedRegistries list
+	allowedRegs := imgcfg.Spec.RegistrySources.AllowedRegistries
+	if len(imgcfg.Status.InternalRegistryHostname) > 0 {
```

Comment:
Shouldn’t this happen only if `len(allowedRegs) > 0`? Otherwise this turns an “allow all” configuration into “allow only the internal registry”, with no way to achieve “allow all”.